tracing-subscriber/src/registry/sharded.rs

```diff
@@ -235,7 +235,11 @@ impl Subscriber for Registry {
     fn clone_span(&self, id: &span::Id) -> span::Id {
         let span = self
             .get(&id)
-            .unwrap_or_else(|| panic!("tried to clone {:?}, but no span exists with that ID", id));
+            .unwrap_or_else(|| panic!(
```

Comment:
I wonder if we should also consider changing this function (and some other similar ones) to only panic in debug mode, and just perform no operation in release mode (maybe printing a warning?). This way, if the user code does the wrong thing, it will just result in potentially-missing trace data, rather than crashing the program in production.

I'm on the fence about this, since it might make this kind of bug harder to uncover. What do you think?

Reply:
I would *much* rather have a crash to debug than have mysteriously missing trace data (that might make it difficult or confounding to debug some other issue). Please keep the panic.

Reply:
Okay, fair enough. I feel like others have had conflicting opinions in the past ("our software absolutely mustn't crash in production, and it's fine if we lose diagnostics instead"), so I've wondered about trying to make this policy configurable...but that could introduce a lot of complexity.

Reply:
@hawkw I think the planned fix in future tracing is the right long-term solution, which will void this at compile time. In the short-term, if I have a risk of a run-time issue I'd rather have it caught at compile time.

I wonder: is there any way to make this panic *always* trigger, such as by putting it unconditionally in the relevant trait impl rather than only if the Id isn't found?